common/app/services/S3.scala

```diff
@@ -1,168 +1,129 @@
+    put(key: String, value: String, contentType: String, PUBLIC_READ)
   }
 
   def putPrivate(key: String, value: String, contentType: String): Unit = {
-    put(key: String, value: String, contentType: String, Private)
+    put(key: String, value: String, contentType: String, PRIVATE)
   }
 
   def putPrivateGzipped(key: String, value: String, contentType: String): Unit = {
-    putGzipped(key, value, contentType, Private)
-  }
-
-  private def putGzipped(
-      key: String,
-      value: String,
-      contentType: String,
-      accessControlList: CannedAccessControlList,
-  ): Unit = {
-    lazy val request = {
-      val metadata = new ObjectMetadata()
-
-      metadata.setCacheControl("no-cache,no-store")
-      metadata.setContentType(contentType)
-      metadata.setContentEncoding("gzip")
-
-      val valueAsBytes = value.getBytes("UTF-8")
-      val os = new ByteArrayOutputStream()
-      val gzippedStream = new GZIPOutputStream(os)
-      gzippedStream.write(valueAsBytes)
-      gzippedStream.flush()
-      gzippedStream.close()
-
-      metadata.setContentLength(os.size())
-
-      new PutObjectRequest(bucket, key, new ByteArrayInputStream(os.toByteArray), metadata)
-        .withCannedAcl(accessControlList)
-    }
+    val os = new ByteArrayOutputStream()
+    val gzippedStream = new GZIPOutputStream(os)
+    gzippedStream.write(value.getBytes(UTF_8))
+    gzippedStream.flush()
+    gzippedStream.close()
+
+    val request = PutObjectRequest
+      .builder()
+      .bucket(bucket)
+      .key(key)
+      .acl(PRIVATE)
+      .cacheControl("no-cache,no-store")
+      .contentType(contentType)
+      .contentEncoding("gzip")
+      .build()
 
-    try {
-      client.foreach(_.putObject(request))
-    } catch {
-      case e: Exception =>
-        throw e
-    }
+    client.putObject(request, RequestBody.fromBytes(os.toByteArray))
   }
 
-  private def put(key: String, value: String, contentType: String, accessControlList: CannedAccessControlList): Unit = {
-    val metadata = new ObjectMetadata()
-    metadata.setCacheControl("no-cache,no-store")
-    metadata.setContentType(contentType)
-    metadata.setContentLength(value.getBytes("UTF-8").length)
-
-    val request =
-      new PutObjectRequest(bucket, key, new StringInputStream(value), metadata).withCannedAcl(accessControlList)
+  private def put(key: String, value: String, contentType: String, accessControlList: ObjectCannedACL): Unit = {
+    val request = PutObjectRequest
+      .builder()
+      .bucket(bucket)
+      .key(key)
+      .acl(accessControlList)
+      .cacheControl("no-cache,no-store")
+      .contentType(contentType)
+      .build()
 
-    try {
-      client.foreach(_.putObject(request))
-    } catch {
-      case e: Exception =>
-        throw e
-    }
+    client.putObject(request, RequestBody.fromString(value, UTF_8))
   }
 }
 
 object S3 extends S3 {
+  val presigner = S3Presigner.create
```

Comment:
This was a mistake! It leaves the `S3Presigner` setup without explictly setting its region or credentials - on some machines this may work in dev, but in others it won't.

This was a last-minute bit of work we did, adapting for this change that had been merged since we started the migration:

* https://github.com/guardian/frontend/pull/28000

...it's fixed with this PR:

* https://github.com/guardian/frontend/pull/28083